compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -465,7 +442,18 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 self.lower_generics(generics, ImplTraitContext::disallowed()),
                 self.lower_param_bounds(bounds, ImplTraitContext::disallowed()),
             ),
-            ItemKind::MacroDef(..) | ItemKind::MacCall(..) => {
+            ItemKind::MacroDef(MacroDef { ref body, macro_rules }) => {
+                let body = P(self.lower_mac_args(body));
+
+                hir::ItemKind::Macro(hir::MacroDef {
+                    ident: *ident,
+                    vis: *vis,
+                    def_id: hir_id.expect_owner(),
+                    span,
```

Comment:
`def_id`, `ident`, `vis` and `span` are already in the `hir::Item`. `hir::ItemKind::Macro` only needs to contain the `ast::MacroDef`.

Reply:
I tried to do that, but I bumped into problems with intravisit::Visitor::visit_macro_def, which needs to receive a hir::MacroDef. I couldn't figure out how to make the lifetimes work for that if the MacroDef didn't already exist. Note that we also support, for instance, Visitor::visit_fn, so I'm not sure that Visitor::visit_macro_def should be considered redundant with Visitor::visit_item.

Reply:
There are only two uses of `visit_macro_def` left: `rustc_passes::{hir_stats, stability}`. I think the function can be removed altogether.

Reply:
Done!